### PR TITLE
Small stuff

### PR DESCRIPTION
**Pull request by nouiz**
- Make debugprint print the clients information
- doc
- error message
- simplification

### CODE REVIEW

File: theano/compile/debugmode.py [hunk 1]
```diff
@@ -512,7 +512,8 @@ def debugprint(r, prefix='', depth=-1, done=None, print_type=False,
                print_view_map=False, order=None, ids='CHAR',
                stop_on_name=False, prefix_child=None,
                scan_ops=None, profile=None,
-               scan_inner_to_outer_inputs=None, smap=None):
+               scan_inner_to_outer_inputs=None, smap=None,
+               used_ids=None):
```
Comment (lamblin) [inline theano/compile/debugmode.py]: Add doc for that new parameter.

File: theano/tensor/opt.py [hunk 2]
```diff
@@ -1578,7 +1578,7 @@ def dimshuffled_alloc(i):
 def local_fill_sink(node):
     """
     f(fill(a, b), fill(c, d), e) -> fill(a, fill(c, f(b, d, e)))
-
+    f(fill(a, b), fill(c, d), e) -> fill(c, fill(a, f(b, d, e)))
```
Comment (lamblin) [inline theano/tensor/opt.py]: Is the new line supposed to replace the previous one?

File: theano/tests/test_printing.py [hunk 3]
```diff
@@ -270,6 +271,26 @@ def test_debugprint():
...
+    s = s.getvalue()
+    # The additional white space are needed!
+    reference = '\n'.join([
+"Elemwise{add,no_inplace} [id A] ''   0 clients:[('output', ''), ('[id C]', 1)]",
+" |A [id D]",
+" |B [id E]",
+"Elemwise{sub,no_inplace} [id C] ''   1",
+" |Elemwise{add,no_inplace} [id A] ''   0 clients:[('output', ''), ('[id C]', 1)]",
+" |D [id F]",
+    ]) + '\n'
```
Comment (lamblin) [inline theano/tests/test_printing.py]: Are you changing the default appearance of `debugprint`? If so, the documentation and examples should be updated everywhere.

Comment (nouiz) [inline theano/tests/test_printing.py]: Right. I don't want to change the default, so I updated this PR to make this an option.

**Pull request merged**